src/RestApi/StoreApi/Controllers/CartOrder.php

```diff
@@ -0,0 +1,362 @@
+<?php
+/**
+ * Cart Order controller.
+ *
+ * @internal This API is used internally by Blocks--it is still in flux and may be subject to revisions.
+ * @package WooCommerce/Blocks
+ */
+
+namespace Automattic\WooCommerce\Blocks\RestApi\StoreApi\Controllers;
+
+defined( 'ABSPATH' ) || exit;
+
+use \WP_Error as RestError;
+use \WP_REST_Server as RestServer;
+use \WP_REST_Controller as RestController;
+use \WP_REST_Response as RestResponse;
+use \WP_REST_Request as RestRequest;
+use \WC_REST_Exception as RestException;
+use Automattic\WooCommerce\Blocks\RestApi\StoreApi\Schemas\OrderSchema;
+
+/**
+ * Cart Order API.
+ *
+ * Creates orders based on cart contents.
+ */
+class CartOrder extends RestController {
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc/store';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'cart/order';
+
+	/**
+	 * Schema class instance.
+	 *
+	 * @var object
+	 */
+	protected $schema;
+
+	/**
+	 * Setup API class.
+	 */
+	public function __construct() {
+		$this->schema = new OrderSchema();
+	}
```

Comment:
I know this isn't a new introduced pattern, and I'm not asking to change it here. But at some point we should switch away from this and pass in dependencies via the constructor.